ext/objspace/objspace_dump.c

```diff
@@ -403,6 +419,27 @@ dump_object(VALUE obj, struct dump_config *dc)
             dump_append(dc, rb_enc_name(rb_enc_from_index(ENCODING_GET(obj))));
             dump_append(dc, "\"");
         }
+
+        dump_append(dc, ", \"coderange\":\"");
+        switch (RB_ENC_CODERANGE(obj)) {
+          case RUBY_ENC_CODERANGE_UNKNOWN:
+            dump_append(dc, "unknown");
+            break;
+          case RUBY_ENC_CODERANGE_7BIT:
+            dump_append(dc, "7bit");
+            break;
+          case RUBY_ENC_CODERANGE_VALID:
+            dump_append(dc, "valid");
+            break;
+          case RUBY_ENC_CODERANGE_BROKEN:
+            dump_append(dc, "broken");
+            break;
+        }
+        dump_append(dc, "\"");
+
+        if (RB_ENC_CODERANGE(obj) == RUBY_ENC_CODERANGE_BROKEN)
+            dump_append(dc, ", \"broken\":true");
```

Comment:
Yeah, it's a bit extra, but it was there before, so removing it would be somewhat of a breaking change.

> is it customary to leave off the false case?

Yes, several other attributes are like that. Heap dumps can be huge, so it's actually a good thing not to include redundant information.